### ISSUE TITLE
[no-use-before-define] does not working when using optional chaining

### ISSUE THREAD

**Issue by nodece** [typescript-eslint/typescript-eslint#1181]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/<rule>": ["<setting>"]
  }
}
```

```TS
const rm = this.model.model.get('g')?.get('g')?.rm;
```

**Expected Result**
No warning
**Actual Result**
```
error  'rm' was used before it was defined                                             @typescript-eslint/no-use-before-define
```
**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.6.1` |
| `@typescript-eslint/parser`        | `2.6.1` |
| `TypeScript`                       | `3.7.0` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `10.16.0` |
| `yarn`                              | `1.19.1` |

**Comment by bradzacher** [comment]
duplicate of #1104

Please use the search next time.
